src/osx/cocoa/colour.mm

```diff
@@ -13,39 +13,103 @@
 
 #include "wx/osx/private.h"
 
-// Helper function to avoid writing too many casts in wxColour ctor.
-static inline wxColour::ChannelType NSColorChannelToWX(CGFloat c)
+class wxNSColorRefData : public wxColorRefData
 {
-    return static_cast<wxColour::ChannelType>(c * 255 + 0.5);
+public:
+    wxNSColorRefData(WX_NSColor color);
+    
+    wxNSColorRefData(const wxNSColorRefData& other);
+
+    virtual ~wxNSColorRefData();
+    
+    virtual bool IsOk() const wxOVERRIDE { return m_nsColour != NULL; }
+    
+    virtual CGFloat Red() const wxOVERRIDE;
+    virtual CGFloat Green() const wxOVERRIDE;
+    virtual CGFloat Blue() const wxOVERRIDE;
+    virtual CGFloat Alpha() const wxOVERRIDE;
+    
+    CGColorRef GetCGColor() const wxOVERRIDE;
+    
+    virtual wxColorRefData* Clone() const wxOVERRIDE { return new wxNSColorRefData(*this); }
+    
+    virtual WX_NSColor GetNSColor() const wxOVERRIDE;
+private:
+    WX_NSColor m_nsColour;
+    
+    wxDECLARE_NO_ASSIGN_CLASS(wxNSColorRefData);
+};
+
+wxNSColorRefData::wxNSColorRefData(WX_NSColor color)
+{
+    m_nsColour = [color retain];
 }
 
-wxColour::wxColour(WX_NSColor col)
+wxNSColorRefData::wxNSColorRefData(const wxNSColorRefData& other)
+{
+    m_nsColour = [other.m_nsColour retain];
+}
+
+wxNSColorRefData::~wxNSColorRefData()
+{
+    [m_nsColour release];
+}
+
+WX_NSColor wxNSColorRefData::GetNSColor() const
+{
+    return m_nsColour;
+}
+
+CGFloat wxNSColorRefData::Red() const
+{
+    if ( NSColor* colRGBA = [m_nsColour colorUsingColorSpaceName:NSCalibratedRGBColorSpace] )
+        return [colRGBA redComponent];
+    
+    return 0.0;
```

Comment:
Is it expected for `NSColor::colorUsingColorSpaceName:` to fail? If not, we should assert here.

Or maybe we should fall back to something else? But just returning wrong colour silently looks like a bad idea...